Billbee.Api.Client.Test/EndPointIntegrationTests/CustomerAddressesEndPointIntegrationTest.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Billbee.Api.Client.Model;
+using Billbee.Api.Client.Test.EndPointIntegrationTests.Helpers;
+
+namespace Billbee.Api.Client.Test.EndPointIntegrationTests.Helpers
+{
+    public static partial class TestData
+    {
+        public static CustomerAddress CustomerAddress =>
+            new CustomerAddress
+            {
+                FirstName = "John",
+                LastName = "Doe",
+                Street = "Mustergasse",
+                Housenumber = "1",
+                Zip = "12345",
+                City = "Musterstadt",
+                AddressType = 1,
+                CountryCode = "DE",
+                CustomerId = 0
+            };
+        
+        public static CustomerAddress GetCustomerAddress(long? customerId)
+        {
+            var address = CustomerAddress;
+            address.CustomerId = customerId;
+            return address;
+        }
+    }
+}
+
+namespace Billbee.Api.Client.Test.EndPointIntegrationTests
+{
+    [TestClass]
+    public class CustomerAddressesEndPointIntegrationTest
+    {
+#pragma warning disable CS8618
+        public TestContext TestContext { get; set; }
+#pragma warning restore CS8618
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            IntegrationTestHelpers.CheckAccess(TestContext.ManagedType, TestContext.ManagedMethod);
+        }
+
+        [TestMethod]
+        [RequiresApiAccess]
+        public void CustomerAddresses_GetCustomerAddresses_IntegrationTest()
+        {
+            var result = CrudHelpers.GetAll(() => IntegrationTestHelpers.ApiClient.CustomerAddresses.GetCustomerAddresses(1, 5));
+            Assert.IsNotNull(result);
+            Assert.IsNotNull(result.Data);
+        }
+
+        [TestMethod]
+        [RequiresApiAccess]
+        public void CustomerAddresses_GetCustomerAddress_IntegrationTest()
+        {
+            var customer =
+                CrudHelpers.CreateApiResult(c => IntegrationTestHelpers.ApiClient.Customer.AddCustomer(c),
+                    TestData.Customer).Data;
+
+            var customerAddress =
+                CrudHelpers.CreateApiResult(
+                    a => IntegrationTestHelpers.ApiClient.CustomerAddresses.AddCustomerAddress(a),
+                    TestData.GetCustomerAddress(customer.Id)).Data;
+            Assert.IsNotNull(customerAddress.Id);
+            CrudHelpers.GetOneApiResult<CustomerAddress>(
+                (id) => IntegrationTestHelpers.ApiClient.CustomerAddresses.GetCustomerAddress(id),
+                customerAddress.Id.Value);
+        }
+
+        [TestMethod]
+        [RequiresApiAccess]
+        public void CustomerAddresses_AddCustomerAddress_IntegrationTest()
+        {
+            CustomerAddresses_GetCustomerAddress_IntegrationTest();
```

Comment:
Warum ist das hier der gleiche Test?

Reply:
Um keine Produktiv-Daten im TestSystem zu ändern werden bei allen Tests immer neue Entities erzeugt, so auch hier in der Methode `CustomerAddresses_GetCustomerAddress_IntegrationTest()`: hier wird zunächst ein Customer erzeugt und dann für diesen Customer einen CustomerAddress. Anschließend wird versucht, diese Adresse wieder über die Api abzurufen, um sicherzustellen, dass sie auch tatsächlich angelegt wurde.
In diesem Workflow ist deshalb der TestCode für das Testen von `GetCustomerAddress()` und `AddCustomerAddress()` derselbe.